src/line_launcher/mod.rs

```diff
@@ -20,6 +20,44 @@ use playing_state::PlayingState;
 mod progression_state;
 use progression_state::ProgressionState;
 
+enum DurationBetweenSixteenthNotes {
+    Uninitialized,
+    PartiallyInitialized {
+        last_timestamp: SystemTime,
+    },
+    Initialized {
+        last_timestamp: SystemTime,
+        last_duration: Duration,
+    },
+}
+
+impl DurationBetweenSixteenthNotes {
+    pub fn process_beat_message(&mut self, _beat_message: &BeatNumber) {
```

Comment:
It'd feel cleaner to me to have this type be immutable, so this function would return a new `DurationBetweenSixteenthNotes` rather than mutating the existing one.

Reply:
Ok updated to make this return a new `DurationBetweenSixteenthNotes`